app/src/features/Batches/BatchResourceDialog.tsx

```diff
@@ -127,15 +127,18 @@ const BatchResourceDialog = ({
   };
 
   const handleSelectAllClick = () => {
-    if (selectedResourceIds.length === 0 && displayedResources) {
-      setSelectedResourceIds(displayedResources.map((resource) => resource.id));
+    if (displayedResourcesIds && selectedResourceIds.length === 0) {
+      setSelectedResourceIds(displayedResourcesIds);
     } else if (areAllDisplayedResourcesSelected || isCheckboxIndeterminate) {
-      const resourcesToKeep = selectedResourceIds.filter(
+      const resourcesIdsToKeep = selectedResourceIds.filter(
         (id) => !displayedResourcesIds?.includes(id)
       );
-      setSelectedResourceIds(resourcesToKeep);
+      setSelectedResourceIds(resourcesIdsToKeep);
     } else if (displayedResourcesIds) {
-      setSelectedResourceIds(selectedResourceIds.concat(displayedResourcesIds));
+      const newSelectedResourcesIds = selectedResourceIds.concat(
+        displayedResourcesIds
+      );
+      setSelectedResourceIds(newSelectedResourcesIds);
```

Comment:
You can actually merge the 1st and the 3rd if statements. The 1st one is only a particular case of the 3rd one, more general.